lib/rubocop/cop/cop.rb

```diff
@@ -217,6 +217,25 @@ def excluded_file?(file)
         !relevant_file?(file)
       end
 
+      # This method should be overriden when a cop's behavior depends
+      # on state that lives outside of these locations:
+      #
+      #   (1) the file under inspection
+      #   (2) the cop's source code
+      #   (3) the config (eg a .rubocop.yml file)
+      #
+      # For example, some cops may want to look at other parts of
+      # the codebase being inspected to find violations. A cop may
+      # use the presence or absence of file `foo.rb` to determine
+      # whether a certain violation exists in `bar.rb`.
+      #
+      # Overriding this method allows the cop to indicate to RuboCop's
+      # ResultCache system when those external dependencies change,
+      # ie when the ResultCache should be invalidated.
+      def external_dependency_checksum
+        nil
```

Comment:
If you run the cop on `n` files, this method will be called every time, recomputing the checksum `n` times, correct?

Maybe it would be useful for rubocop to have a "run" context that keeps state across files and is accessible from a cop.

Reply:
Good question -- please see https://github.com/rubocop-hq/rubocop/pull/7543. I believe these results should be cached per team/config, so don't need to be recomputed per inspected file.